### PR TITLE
Don't reuse background web contents

### PR DESCRIPTION
**Pull request by sangwoo108**
Reusing contents could make DidFinishLoad() not called.

e.g.
  * https://youtube.com was loaded on the contents
  * Make a new request for http://youtube.com#searching
  * As the youtube.com was already loaded, it just moves scroll to #searching fragment and doesn't reload. As a result, we can't get DidFinishLoad() callback.
  * This make the request never ends.

Resolves https://github.com/brave/brave-browser/issues/28944

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/playlist/browser/playlist_download_request_manager.cc [hunk 1]
```diff
@@ -95,16 +104,24 @@ void PlaylistDownloadRequestManager::FetchPendingRequest() {
...
   DCHECK(callback_for_current_request_.is_null());
   callback_for_current_request_ = std::move(request.callback);
+  DCHECK(callback_for_current_request_)
+      << "Empty callback shouldn't be requested";
+  request_start_time_ = base::Time::Now();
 
   if (absl::holds_alternative<std::string>(request.url_or_contents)) {
+    // Start to request on clean slate, so that result won't be affected by
+    // previous page.
+    web_contents_.reset();
```
Comment (simonhong) [inline components/playlist/browser/playlist_download_request_manager.cc:122]: This reset needs here? Can we add `DCHECK(!web_contents);` at the start of `CreateWebContents()` instead?

Comment (sangwoo108) [reply components/playlist/browser/playlist_download_request_manager.cc:122]: Yes, that'd be better 👍

Comment (simonhong) [reply components/playlist/browser/playlist_download_request_manager.cc]: Ah, I mean adding `DCHECK()` to the start of `CreateWebContents();` :)

**Review by simonhong — APPROVED** [review]

File: components/playlist/browser/playlist_download_request_manager.cc (hunk 1 above)
Comment (sangwoo108) [reply components/playlist/browser/playlist_download_request_manager.cc]: Oh, yeah. That's even better.

**Review by deeppandya — APPROVED** [review]
LGTM. I have also verified the change on the build.

**Pull request merged**